### DISCUSSION TITLE
Which of the three methods is correct for creating a UserControl with different x:class, using, and namespace?

### DISCUSSION THREAD

**Discussion by chatterzhao** [AvaloniaUI/Avalonia#15797] [Q&A]
![d714de7a50ebb71d94d9a0d6f4515da2](https://github.com/AvaloniaUI/Avalonia/assets/11829944/0072bc39-5fc5-4405-b9d6-d81346e99e6f)

dotnet new avalonia.usercontrol -o Views -n UserControl3  --namespace ToDoList.Views

Which of the three methods is correct for creating a UserControl with different x:class, using, and namespace?

x:Class="ToDoList.Views.UserControl1" or x:Class="ToDoList.UserControl1"
<img width="592" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/11829944/d67343c2-8f9f-4703-9bd6-3937a4f061b2">
<img width="519" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/11829944/f343ae74-ecc8-48a7-b089-1913e67cc89b">
<img width="591" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/11829944/60d0a27e-8c51-478c-996a-4f76a984877c">

**Comment by timunie** [comment]
I'm sorry but I don't quite understand the question. I'd use the templates with the icons applied but all three options should work. If you adjust namespace or name, remember to update the XAML file accordingly.

**Reply by chatterzhao** [reply]
<img width="716" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/11829944/af3645bf-1107-4696-8aed-a13a01015f03">

Using the three methods displayed in the above screenshot to create a “UserControl”, I obtain different “x:class, namespace, using”. Which one is the correct way? Or is it possible to make the results of these three methods the same?

**Reply by timunie** [reply]
See https://github.com/AvaloniaUI/Avalonia/discussions/15797#discussioncomment-9540039

**Comment by stevemonaco** [comment]
There is no "correct" here, only opinions. Just be consistent with your namespace conventions across the entire project. I use feature folders, so I always need to clean up namespaces to match my convention.

The top templates (plain icon) are from AvaloniaVS. The bottom templates (Avalonia icon) are from avalonia-dotnet-templates. The latter cannot match folder-namespace convention due to https://github.com/AvaloniaUI/avalonia-dotnet-templates/issues/235
